packages/test/test-end-to-end-tests/src/test/messageSize.spec.ts

```diff
@@ -535,4 +535,96 @@ describeNoCompat("Message size", (getTestObjectProvider) => {
 			});
 		});
 	});
+
+	describe("Payload size on the wire", () => {
+		let totalPayloadSizeInBytes = 0;
+
+		const assertPayloadSize = (totalMessageSizeInBytes: number): void => {
+			// Expecting the message size on the wire should have
+			// at most 35% extra from stringification and envelope overhead.
+			const overheadRatio = 1.35;
+			assert.ok(
+				totalPayloadSizeInBytes < overheadRatio * totalMessageSizeInBytes,
+				`Message size on the wire, ${totalPayloadSizeInBytes} is larger than expected ${
+					overheadRatio * totalMessageSizeInBytes
+				}, after sending ${totalMessageSizeInBytes} bytes`,
+			);
+		};
+
+		const setup = async (containerConfig: ITestContainerConfig) => {
+			await setupContainers(containerConfig);
+			totalPayloadSizeInBytes = 0;
+			localContainer.deltaManager.outbound.on(
+				"push",
+				(messages) => (totalPayloadSizeInBytes += JSON.stringify(messages).length),
+			);
+		};
+
+		const compressionRatio = 0.1;
+		const badCompressionRatio = 1;
+		describe("Check batch size on the wire", () =>
+			[
+				{
+					messagesInBatch: 1,
+					messageSize: 1024,
+					expectedSize: 1 * 1024,
+					payloadGenerator: generateStringOfSize,
+				}, // One small uncompressed message
+				{
+					messagesInBatch: 3,
+					messageSize: 1024,
+					expectedSize: 3 * 1024,
+					payloadGenerator: generateStringOfSize,
+				}, // Three small uncompressed messages
+				{
+					messagesInBatch: 1,
+					messageSize: 1025 * 1024,
+					expectedSize: compressionRatio * 1025 * 1024,
```

Comment:
Does compression automatically kick in when message size is above a threshold?

Reply:
Yes. The threshold is here https://github.com/microsoft/FluidFramework/pull/14587/files#diff-cf13339c0434ab032e25d00d0611fcb811f01516c2f1a2841d1b1def776dc44eR306

Reply:
I see.
nit: Maybe declare the threshold as a const and use it to determine whether the expected size should be compressed or not so that everything is in sync.